iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2312,6 +2509,107 @@ TEST(String10, AppendTooLargeStringLiteralResultsInTruncatedString)
     EXPECT_THAT(testString.c_str(), StrEq("Live long "));
 }
 
+TYPED_TEST(stringTyped_test, AppendEmptyStdStringWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6b3fb31a-a92b-4013-ba61-12b4846e8593");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    this->testSubject = "M";
+    std::string testStdString;
+    this->testSubject.append(TruncateToCapacity, testStdString);
+    EXPECT_THAT(this->testSubject.capacity(), Eq(STRINGCAP));
+    EXPECT_THAT(this->testSubject.size(), Eq(1U));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq("M"));
+}
+
+TYPED_TEST(stringTyped_test, AppendStdStringToEmptyStringResultsInConcatenatedString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2eebdc68-c495-4531-bb66-ee4ede8d86e3");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    std::string testStdString = "M";
+    this->testSubject.append(TruncateToCapacity, testStdString);
+    EXPECT_THAT(this->testSubject.capacity(), Eq(STRINGCAP));
+    EXPECT_THAT(this->testSubject.size(), Eq(1U));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq("M"));
+}
+
+TYPED_TEST(stringTyped_test, AppendStdStringResultsInConcatenatedString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "7159ef46-f441-4cc7-8eff-46b5f3d33597");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<STRINGCAP + 5U> sut = "P";
+    std::string testStdString("icard");
+    sut.append(TruncateToCapacity, testStdString);
+    EXPECT_THAT(sut.capacity(), Eq(STRINGCAP + 5U));
+    EXPECT_THAT(sut.size(), Eq(6U));
+    EXPECT_THAT(sut.c_str(), StrEq("Picard"));
+}
+
+TYPED_TEST(stringTyped_test, AppendTooLargeStdStringResultsInTruncatedString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "76a4752d-282f-47ad-87eb-7c8aab982c0c");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    this->testSubject = "M";
+    std::string testStdString(STRINGCAP + 1U, 'M');
+    this->testSubject.append(TruncateToCapacity, testStdString);
+    EXPECT_THAT(this->testSubject.capacity(), Eq(STRINGCAP));
+    EXPECT_THAT(this->testSubject.size(), Eq(STRINGCAP));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(testStdString.substr(0, STRINGCAP)));
+}
+
+/// @note string& append(TruncateToCapacity_t, char c) noexcept
+TYPED_TEST(stringTyped_test, AppendNullCharWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "c20f1797-2797-4f4a-9acf-7bd1b9518c3d");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<STRINGCAP + 1U> sut = "M";
+    sut.append(TruncateToCapacity, '\0');
+    EXPECT_THAT(sut.capacity(), Eq(STRINGCAP + 1U));
+    EXPECT_THAT(sut.size(), Eq(2U));
+    EXPECT_THAT(sut.c_str(), StrEq("M"));
```

Comment:
Shouldn't we check as well that the memory returned by `c_str()` contains `M\0`? We could verify that with `memcmp`.